### PR TITLE
ENT-3482:Add functionality to save logo in EnterpriseCustomerBrandingConfiguration at only one location

### PR DESCRIPTION
**Pull request by sameenfatima78**
JIRA: https://openedx.atlassian.net/browse/ENT-3482

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [x] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Comment by HammadAhmadWaqas** [comment]
@sameenfatima78 could we add a test to support our changes?
ideally, that test should fail without your changes and pass after implementing your changes.

**Review by HammadAhmadWaqas — COMMENTED** [review]
@sameenfatima78 could we add a test to support our changes?
ideally, that test should fail without your changes and pass after implementing your changes.

**Review by saleem-latif — APPROVED** [review]
Just one question otherwise looks good 👍

File: enterprise/signals.py [hunk 1]
```diff
@@ -99,17 +99,24 @@ def _complete_user_enrollment():
...
+def save_logo_file(sender, instance, **kwargs):            # pylint: disable=unused-argument
     """
-    Update logo name and path to replace None with instance id.
+    Now that the object is instantiated and instance.id exists, save the image at correct path and re-save the model.
     """
-    filename = instance.logo.name
-    if kwargs['created'] and 'None' in filename:
-        extension = os.path.splitext(filename)[1].lower()
-        instance_file = instance.logo.file
-        new_name = 'enterprise/branding/' + str(instance.id) + '/' + str(instance.id) + '_logo' + extension
```
[the remark this answers is not in the document]
Comment (sameenfatima78) [reply enterprise/signals.py:111]: @saleem-latif No, this code was actually added by me to fix ENT-3406. It was added to re-save the instance at the correct path once the model object was created. But now we are only saving the image once after post_save and the path to upload the image is already given [here](https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L1129) already.

File: enterprise/signals.py [hunk 2]
```diff
@@ -99,17 +99,24 @@ def _complete_user_enrollment():
...
-        instance.logo.save(new_name, instance_file)
+    if kwargs['created'] and hasattr(instance, _UNSAVED_FILEFIELD):
+        instance.logo = getattr(instance, _UNSAVED_FILEFIELD)
+        instance.__dict__.pop(_UNSAVED_FILEFIELD)
```
Comment (HammadAhmadWaqas) [inline enterprise/signals.py:119]: can we use `delattr` builtin function here?

Comment (sameenfatima78) [reply enterprise/signals.py]: Good suggestion. I have incorporated that.

File: tests/test_models.py [hunk 3]
```diff
@@ -782,6 +782,23 @@ def test_logo_path(self, file_exists, delete_called):
...
+        branding_config = EnterpriseCustomerBrandingConfiguration(
+            enterprise_customer=factories.EnterpriseCustomerFactory(),
+            logo="logo.png"
+        )
+        storage_mock = mock.MagicMock(spec=Storage, name="StorageMock")
+        with mock.patch("django.core.files.storage.default_storage._wrapped", storage_mock):
+            branding_config.save()
+            config_id = branding_config.id
+            path = logo_path(branding_config, branding_config.logo.name)
+            expected_path = "enterprise/branding/"+str(config_id)+"/"+str(config_id)+"_logo.png"
```
Comment (HammadAhmadWaqas) [inline tests/test_models.py:799]: I think we should assert the saved path of the image with the expected path from the logo_path.
```
saved_path = branding_config.logo.path
expected_path = logo_path(branding_config, branding_config.logo.name)
self.assertEqual(saved_path, expected_path)
```

**Review by HammadAhmadWaqas — COMMENTED** [review]
minor suggestion, otherwise good.

File: tests/test_models.py (hunk 3 above)
Comment (sameenfatima78) [reply tests/test_models.py:799]: I can't check the exact path as I'm mocking Storage. All I need to test is that the path has the correct instance id in it instead of None.

Comment (HammadAhmadWaqas) [reply tests/test_models.py:799]: I didn't get why we have to mock storage for this test?

Comment (sameenfatima78) [reply tests/test_models.py:799]: Let's catch up on Slack

**Review by HammadAhmadWaqas — APPROVED** [review]
